lib/util.js

```diff
@@ -779,14 +764,15 @@ function formatCollectionIterator(ctx, value, recurseTimes, visibleKeys, keys) {
 }
 
 function formatPromise(ctx, value, recurseTimes, visibleKeys, keys) {
-  var output = [];
-  var internals = inspectPromise(value);
-  if (internals.status === 'pending') {
+  const output = [];
+  const [state, result] = binding.getPromiseDetails(value);
+
+  if (state === binding.kPending) {
     output.push('<pending>');
   } else {
     var nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-    var str = formatValue(ctx, internals.value, nextRecurseTimes);
-    if (internals.status === 'rejected') {
+    var str = formatValue(ctx, result, nextRecurseTimes);
+    if (state === binding.kRejected) {
```

Comment:
Errr, this allows you to get the actual result object e.g. an error without going through the Debug API? I was told that was not possible to do synchronously.